portafly/src/pages/Overview.tsx

```diff
@@ -1,32 +1,90 @@
-import React from 'react'
+import React, { useState, useContext } from 'react'
 import {useA11yRouteChange, useDocumentTitle} from 'components'
 import {
   PageSection,
   TextContent,
   Title,
   Text,
   Card,
-  CardBody
+  CardBody,
+  Modal,
+  Button
 } from '@patternfly/react-core'
 
+import moment from 'moment'
+import { I18nContext } from 'i18n/I18nProvider'
+import { 
+  OVERVIEW as STRINGS_ENG,
+  COMMONS as COMMONS_ENG
+} from 'i18n/locales/en/strings'
+import { 
+  OVERVIEW as STRINGS_JAP,
+  COMMONS as COMMONS_JAP
+} from 'i18n/locales/jp/strings'
+
 const Overview: React.FunctionComponent = ({children}) => {
+  const [strings, setStrings] = useState(STRINGS_ENG)
+  const [commonStrings, setCommonStrings] = useState(COMMONS_ENG)
+  const [locale, setLocale] = useState('en')
+  const polyglot = useContext(I18nContext)
+  polyglot.locale(locale)  
+  polyglot.extend({ ...strings, ...commonStrings })
+
+  const onClick = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    setStrings(event.currentTarget.value === 'en' ? STRINGS_ENG : STRINGS_JAP)
+    setCommonStrings(event.currentTarget.value === 'en' ? COMMONS_ENG : COMMONS_JAP)
+    setLocale(event.currentTarget.value)
+  }
+
   useA11yRouteChange()
-  useDocumentTitle('Overview')
+  useDocumentTitle(polyglot.t('page_title'))
```

Comment:
We could import the `t` only and make it less verbose since it's going to be.. everywhere
```suggestion
  useDocumentTitle(t('page_title'))
```

Reply:
In lines 30 and 31 we are passing the `locale` (language) and `phrases`. If we only import the `t` method we can't do that.
We are not adding a second language shortly, so `locale`is not a big problem. But being able to pass the `phrases` is important.

Reply:
maybe we could import them separately, spreading the import, no?

Reply:
You could even import both, I don't se the problem. And initialization code such as `locale` could be move to an initialization file such as the Context (repeating myself https://github.com/3scale/porta/pull/1776/files#r393571503)

Reply:
I moved it to a setup file